src/CGIHandler.cpp

```diff
@@ -12,10 +10,110 @@ CGIHandler::~CGIHandler()
 {
 }
 
+CGIHandler &CGIHandler::operator=(const CGIHandler &other)
+{
+	if (this != &other)
+	{									   // Protect against self-assignment
+		ARequestHandler::operator=(other); // Call the base class assignment operator
+										   // Copy or assign other members of CGIHandler if necessary
+	}
+	return *this;
+}
+
 HTTPResponse CGIHandler::handleRequest(const HTTPRequest &request)
```

Comment:
The CGIHanlder has an handleRequest that call an handleCGIRequest function, without doing that much around it. Why can't have this method the content of handleCGIRequest?

Reply:
Yes I know, this was due to inheritance and the virtual type of the method, I tried to follow the base object you designed, and at the moment it is too complicated, I'll see how to make it better.

Reply:
Exactly the problem is that subclasses of the base class are expected to provide an implementation for a virtual function., so I was kind of forced to do it.
`ARequestHandler` is the base class and at the moment we use it only for `class CGIHandler : public ARequestHandler`, I kept the code as you designed because I thought you had a more global idea when designing it but now I am thinking that perhaps we don't need `ARequestHandler`, what do you think @552020  ?

Reply:
This is also related to https://github.com/dantol29/webserver/pull/117#discussion_r1568391340

Reply:
The handleRequest function is the virtual method you need to implement. It's not necessary to have an handleCGIRequest method, you can copy that code in the handleRequest method. The handleRequest method will be recognisible as the handing a CGI request cause it will be called by a cgiHanlder object. 

I mean it abstract class it's not necessary (almost anything is). And you can for sure delete it, if you don't need it and you want to design the request handling objects in another way. It seems to me though that you somehow by the way you are using it not fully understanding at least the idea behind its implementation.

Reply:
> It's not necessary to have an handleCGIRequest method

It is, if it is in the base class (otherwise the compiler outputs an error) that's what I meant.

I might be wrong but I think I understand the inheritance, my point is that if we have only one subclass (CGIHandler) then we don't need a base class (ARequestHandler), unless we plan to use it as a base for HTTPRequestHandler (which doesn't seem necessary). 

Additionally I realize now [with this graph](https://miro.com/app/board/uXjVKbFlOrg=/?moveToWidget=3458764586007390181&cot=14) that CGIHandler is rather on the response side than the request side


> And you can for sure delete it, if you don't need it and you want to design the request handling objects in another way.
I would delete it yes if you don't mind. But the request I think is fine the way it currently is, or eventually with a parser object, and as said above I think the CGI is more response side than request side

Reply:
> > It's not necessary to have an handleCGIRequest method
> 
> It is, if it is in the base class (otherwise the compiler outputs an error) that's what I meant.

you need the handleRequest method not the hanldeCGIRequest method, you would need to implement the abstract method. 
> 
> I might be wrong but I think I understand the inheritance, my point is that if we have only one subclass (CGIHandler) then we don't need a base class (ARequestHandler), unless we plan to use it as a base for HTTPRequestHandler (which doesn't seem necessary).
> 
> Additionally I realize now [with this graph](https://miro.com/app/board/uXjVKbFlOrg=/?moveToWidget=3458764586007390181&cot=14) that CGIHandler is rather on the response side than the request side
> 
> > And you can for sure delete it, if you don't need it and you want to design the request handling objects in another way.
> 
> I would delete it yes if you don't mind. But the request I think is fine the way it currently is, or eventually with a parser object, and as said above I think the CGI is more response side than request side

Go for it!